p2panda-core/src/extensions.rs

```diff
@@ -61,8 +28,8 @@ use crate::Header;
 
 /// Trait definition of a single header extension type.
 pub trait Extension<T>: Extensions {
-    /// Extract the value of an extension based on it's type.
-    fn extract(&self) -> Option<T> {
+    /// Extract the extension value from a header.
+    fn extract(_header: &Header<Self>) -> Option<T> {
         None
     }
 }
```

Comment:
We're back to one `extract` method on `Extension` which takes a header as argument.

Reply:
I wonder if we want to actually return `T` directly now, without wrapping in an `Option`. If a field is actually optional, this could be taken care of in your custom extension type definition.

Reply:
Hmm, although I'm not sure, the fact that extensions are optional on header themselves effects this return value too.

Reply:
> Hmm, although I'm not sure, the fact that extensions are optional on header themselves effects this return value too.

Yeah right, it's a good way to check if a value was given in the header or not (even if mandatory), so the application can accordingly consider an operation invalid.